buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/generator/ClassJsonElementDeserializerGenerator.kt

```diff
@@ -251,7 +243,7 @@ class ClassDeserializerGenerator(
         rootTypeName: String
     ) {
         val opt = if (nullable) "?" else ""
-        beginControlFlow("$assignee = $getter$opt.toString()$opt.let")
+        beginControlFlow("$assignee = ($getter as$opt JsonObject)$opt.let")
```

Comment:
same: I think it is better to use `asJsonObject` here, different exception will be thrown. And below on the L268 as well.